test/unit/client-side-encryption/state_machine.test.ts

```diff
@@ -251,6 +251,41 @@ describe('StateMachine', function () {
       });
     });
 
+    context('when server closed the socket', function () {
+      let server;
+
+      beforeEach(async () => {
+        server = net.createServer(async socket => {
+          socket.end();
+        });
+        server.listen(0);
+        await once(server, 'listening');
+      });
+
+      afterEach(() => {
+        server.close();
+      });
+
+      it('throws a MongoCryptError error', async function () {
+        const stateMachine = new StateMachine({
+          proxyOptions: {
+            proxyHost: 'localhost',
+            proxyPort: server.address().port
+          }
+        } as any);
+
+        const request = new MockRequest(Buffer.from('foobar'), 500);
+        try {
+          await stateMachine.kmsRequest(request);
```

Comment:
```suggestion
        const error = await stateMachine.kmsRequest(request).catch(error => error);
```

By attaching a catch handler to return the error you can remove control flow (like early return) and expect.fail from this test.

Reply:
The current format was chosen for consistency with other tests in this file and to be honest, it looks cleaner than `.catch(error => error)`. Would you be ok to keep it as it?